exec/java-exec/src/main/java/org/apache/drill/exec/ops/FragmentContext.java

```diff
@@ -17,483 +17,216 @@
-    final FragmentHandle handle = getHandle();
-    final String frag = handle != null ? handle.getMajorFragmentId() + ":" + handle.getMinorFragmentId() : "0:0";
-    return frag;
-  }
+  <T> List<T> getImplementationClass(final CodeGenerator<T> cg, final int instanceCount)
+      throws ClassTransformationException, IOException;
 
   /**
-   * Get this fragment's allocator.
-   * @return the allocator
+   * Return the set of execution controls used to inject faults into running
+   * code for testing.
+   *
+   * @return the execution controls
    */
-  @Deprecated
-  public BufferAllocator getAllocator() {
-    if (allocator == null) {
-      logger.debug("Fragment: " + getFragIdString() + " Allocator is NULL");
-    }
-    return allocator;
-  }
+  ExecutionControls getExecutionControls();
 
-  public BufferAllocator getNewChildAllocator(final String operatorName,
-      final int operatorId,
-      final long initialReservation,
-      final long maximumReservation) throws OutOfMemoryException {
-    return allocator.newChildAllocator(
-        "op:" + QueryIdHelper.getFragmentId(fragment.getHandle()) + ":" + operatorId + ":" + operatorName,
-        initialReservation,
-        maximumReservation
-        );
-  }
+  /**
+   * Returns the Drill configuration for this run. Note that the config is
+   * global and immutable.
+   *
+   * @return the Drill configuration
+   */
+  DrillConfig getConfig();
 
-  public boolean isOverMemoryLimit() {
-    return allocator.isOverLimit();
-  }
+  FragmentStats getStats();
 
-  @Override
-  protected CodeCompiler getCompiler() {
-    return context.getCompiler();
-  }
+  CodeCompiler getCompiler();
 
-  public AccountingUserConnection getUserDataTunnel() {
-    Preconditions.checkState(connection != null, "Only Root fragment can get UserDataTunnel");
-    return accountingUserConnection;
-  }
+  Collection<CoordinationProtos.DrillbitEndpoint> getBits();
 
-  public ControlTunnel getControlTunnel(final DrillbitEndpoint endpoint) {
-    return context.getController().getTunnel(endpoint);
-  }
+  CoordinationProtos.DrillbitEndpoint getForemanEndpoint();
 
-  public AccountingDataTunnel getDataTunnel(final DrillbitEndpoint endpoint) {
-    AccountingDataTunnel tunnel = tunnels.get(endpoint);
-    if (tunnel == null) {
-      tunnel = new AccountingDataTunnel(context.getDataConnectionsPool().getTunnel(endpoint), sendingAccountor, statusHandler);
-      tunnels.put(endpoint, tunnel);
-    }
-    return tunnel;
-  }
+  CoordinationProtos.DrillbitEndpoint getEndpoint();
 
-  public IncomingBuffers getBuffers() {
-    return buffers;
-  }
+  Controller getController();
 
-  public OperatorContext newOperatorContext(PhysicalOperator popConfig, OperatorStats stats)
-      throws OutOfMemoryException {
-    OperatorContextImpl context = new OperatorContextImpl(popConfig, this, stats);
-    contexts.add(context);
-    return context;
-  }
+  OperatorCreatorRegistry getOperatorCreatorRegistry();
```

Comment:
Needed only when creating operators, not when executing a fragment. So, network/cluster related.

Rather than continue to tag each item; please review the original interface class where I did this analysis to decide which methods should be in the runtime interface and which are network/cluster/server related (and so I left them in the implementation class.)

Since we have no design doc, perhaps we can at least document these decisions and concepts in the class header comments. (I suppose I should have done that; rereading the comment it clearly lacks this level of detail.)

Reply:
Moved to ExecutorFragmentContext